### ISSUE TITLE
API-API

### ISSUE THREAD

**Issue by jens-ox** [bundesAPI/sofortmassnahmen#68]
Aktuell ist das [API-Register](https://github.com/bundesAPI/apis) nicht automatisiert. Gibt es irgendwelche technischen Hürden, die gegen eine Automatisierung sprechen?

* `name`: `openapi.yaml` -> `info.title`
* `office`: aktuell glaub nicht systematisiert, vielleicht `info.x-office` in den OpenAPI-Specs? 
* `description`: ~`openapi.yaml` -> `info.description`~ GitHub repo description
* `documentationUrl`: ~`CNAME.bund.dev`~ GitHub repo project URL
* `githubUrl`: klar

Am einfachsten wäre wahrscheinlich ein GitHub-Actions-Cronjob in https://github.com/bundesAPI/apis, der folgendes macht:

* Liste öffentlicher Repos von `bundesAPI` laden (via https://github.com/octokit/action.js/)
* Liste nach `*-api` filtern
* Alle OpenAPI-Specs laden und verwursten
* Resultat committen

**Comment by wirthual** [comment]
I like that idea!!

If we make it smart, it should check which entries are already there and not override them. This way we could also easily add the office manually at the beginning. 

My approach would be using a python script for this, but I am also happy to do it in any other language if it's preferred. 

Let me know what you think :)

**Comment by jens-ox** [comment]
I think the easiest would be to use JavaScript for this, because it can be [directly called from GitHub Actions](https://docs.github.com/en/actions/creating-actions/creating-a-javascript-action). Also I'm a JS person and don't know lots of Python, so it would be easier for me 😅

**Comment by wirthual** [comment]
Sounds good to me 😊

Do you want to start a skeleton for it? Happy to contribute in JS as well 👍

**Comment by jens-ox** [comment]
I'll raise a PR tomorrow morning ☺️

**Comment by LilithWittmann** [comment]
Hmm. not sure if we really should use the api description automatically. Or
if there shouldn't be a seperate api description for the API list which is
more user friendly.

Am Fr., 15. Juli 2022 um 16:32 Uhr schrieb wirthual <
***@***.***>:

**Comment by jens-ox** [comment]
Any reason why we can't just make the descriptions provided by the APIs more user-friendly? Also, I found the descriptions (e.g. [Autobahn API](https://github.com/bundesAPI/autobahn-api/blob/main/openapi.yaml#L5)) to be quite good so far ☺️

**Comment by wirthual** [comment]
I think taking the description from the file could be a good starting point (Better than leaving it empty I assume). 

It should always be possible to overwrite it manually without the script overwriting it again.

**Comment by jens-ox** [comment]
> It should always be possible to overwrite it manually without the script overwriting it again.

That would be nice, not sure if it's desirable though. What description is "true" if the description is manually changed in the output file *and* and the source repository?

I think that the source of truth should be kept inside the source repository whenever possible. Overwriting it in the output file should be nothing more than an escape hatch.

**Comment by jens-ox** [comment]
See linked PR in the API repository.

tl;dr:

* overrides can be set in `overrides.json`
* PR is automatically created if any changes exist
* Action takes information from public repos list and each repo's `openapi.yaml`

Next, we should set the deployed Swagger URLs as the project URL for each repo (which is already done for most repos) and add/update repo descriptions. I think those two places are the sanest sources of truth for deploy URL and repo description.
